app/modules/adxl345.c

```diff
@@ -0,0 +1,101 @@
+/*
+ * Driver for Analog Devices ADXL345 accelerometer.
+ *
+ * Code based on BMP085 driver.
+ */
+#include "module.h"
+#include "lauxlib.h"
+#include "platform.h"
+#include "c_stdlib.h"
+#include "c_string.h"
+
+static const uint32_t adxl345_i2c_id = 0;
+static const uint8_t adxl345_i2c_addr = 0x53;
+
+static uint8_t ICACHE_FLASH_ATTR r8u(uint32_t id, uint8_t reg) {
+    uint8_t ret;
+
+    platform_i2c_send_start(id);
+    platform_i2c_send_address(id, adxl345_i2c_addr, PLATFORM_I2C_DIRECTION_TRANSMITTER);
+    platform_i2c_send_byte(id, reg);
+    platform_i2c_send_stop(id);
+    platform_i2c_send_start(id);
+    platform_i2c_send_address(id, adxl345_i2c_addr, PLATFORM_I2C_DIRECTION_RECEIVER);
+    ret = platform_i2c_recv_byte(id, 0);
+    platform_i2c_send_stop(id);
+    return ret;
+}
+
+static int ICACHE_FLASH_ATTR adxl345_init(lua_State* L) {
+
+    uint32_t sda;
+    uint32_t scl;
+    uint8_t  devid;
+
+    if (!lua_isnumber(L, 1) || !lua_isnumber(L, 2)) {
```

Comment:
This check is redundant so drop it; the check integer calls do this anyway. Also use the arg check macro instead of an if predicate and and error: less source line and a better diagnostic for the app programmer.